### PR TITLE
feat: CloudFormation deployment with prebuild artifact

### PR DESCRIPTION
**Pull request by chejimmy**
# Description

Updated the `aws-resources-cfn-template.yaml` to deploy with prebuild artifact

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manually deployed to various account

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Review by square-li — APPROVED** [review]
Ship it! 🚢

File: aws-resources-cfn-template.yaml [hunk 1]
```diff
@@ -56,15 +68,28 @@ Resources:
...
-        # TODO: tighten the authenticated-role-policy policy permissions to require actions only
         - PolicyName: authenticated-role-policy
           PolicyDocument:
             Version: 2012-10-17
             Statement:
               - Sid: SiteWisePermissions
                 Effect: Allow
                 Action:
-                  - 'iotsitewise:*'
+                  - iotsitewise:DescribeAsset
```
Comment (boweihan) [inline aws-resources-cfn-template.yaml:78]: 👍

**Review by boweihan — APPROVED** [review]

File: aws-resources-cfn-template.yaml [hunk 2]
```diff
@@ -93,3 +118,215 @@ Resources:
...
+          const s3 = new aws.S3();
+
+          exports.handler = (event, context) => {
+            // Get the request event info
+            const requestType = event.RequestType;
+            console.log('The event type is: ', requestType);
+            console.log('The resource properties is: ', event.ResourceProperties);
+            const { Body, Bucket, Key } = event.ResourceProperties;
+
+            if (requestType === 'Create' || requestType === 'Update') {
```
Comment (tracy-french) [inline aws-resources-cfn-template.yaml:276]: Is request type of union of string literals? If so, with TypeScript 5, there's a compiler flag for exhaustive switches, which will ensure all cases are handled. Pretty cool!

**Review by tracy-french — APPROVED** [review]
Ship it!

**Comment by tracy-french** [comment]
I shipped it for you 😅. Hope you don't mind!

**Pull request merged**